packages/medusa/src/services/product-collection.js

```diff
@@ -162,6 +162,46 @@ class ProductCollectionService extends BaseService {
     })
   }
 
+  async addProducts(collectionId, productIds) {
+    return this.atomicPhase_(async (manager) => {
+      const productRepo = manager.getCustomRepository(this.productRepository_)
+
+      const collection = await this.retrieve(collectionId)
```

Comment:
A nit but we could be more mem-efficient here with:

```suggestion
      const collection = await this.retrieve(collectionId, { select: ["id"] })
```